### PR TITLE
Enable `allow-same-origin` for iframes in browsable buckets

### PR DESCRIPTION
**Pull request by nl0**
### TODO

- [x] [Changelog](CHANGELOG.md) entry

### CODE REVIEW

File: catalog/app/components/Preview/loaders/Html/Html.tsx [hunk 1]
```diff
@@ -185,3 +195,74 @@ export default function BrowsableLoader({ handle, children }: BrowsableLoaderPro
...
+    () => sign(handle, { ResponseContentType: 'text/html' }),
+    [handle, sign],
+  )
+  // TODO: issue a head request to ensure existence and get storage class
+  return children(
+    AsyncResult.Ok(
+      PreviewData.IFrame({
+        src,
+        modes: [FileType.Html, FileType.Text],
+        sandbox: browsable ? 'allow-scripts allow-same-origin' : 'allow-scripts',
```
Comment (fiskus) [inline catalog/app/components/Preview/loaders/Html/Html.tsx:217]: This is the main change, as I see

File: catalog/app/components/Preview/renderers/IFrame.tsx [hunk 2]
```diff
@@ -11,9 +11,7 @@ const useStyles = M.makeStyles({
 
 function IFrame(props: React.HTMLProps<HTMLIFrameElement>) {
   const classes = useStyles()
-  return (
-    <iframe className={classes.root} title="Preview" sandbox="allow-scripts" {...props} />
```
Comment (fiskus) [inline catalog/app/components/Preview/renderers/IFrame.tsx:15]: And this

**Review by fiskus — APPROVED** [review]

File: docs/Catalog/Preview.md [hunk 3]
```diff
@@ -52,34 +52,83 @@ currently supported.
...
+1. `report.html` is a file in a bucket `example-bucket` that includes a publicly
+   available JS library and custom embedded script.
+2. When opening `report.html` in any view, it is served directly by S3 via a
+   signed HTTPS URL, e.g.
+   `https://example-bucket.s3.region.amazonaws.com/report.html?versionId=...&X-Amz-...`.
+3. All relative media and scripts are rendered in the same iframe relative-path
+   format:
+    * `./img.jpg` is resolved to `/img.jpg`
+    * `script.js` is resolved to `/script.js`
+4. `allow-same-origin` iframe sandbox token disabled,
```
Comment (akarve) [inline docs/Catalog/Preview.md:129]:
```suggestion
4. The `allow-same-origin` iframe sandbox token is disabled,
```

File: docs/Catalog/Preview.md [hunk 4]
```diff
@@ -52,34 +52,83 @@ currently supported.
...
+4. `allow-same-origin` iframe sandbox token enabled,
+   the origin is the **S3 bucket endpoint**
+   (e.g. `https://example-bucket.s3.region.amazonaws.com`),
+   LocalStorage API is available.
+
+### Example with permissive rendering disabled
+
+1. `report.html` is a file in a bucket `example-bucket` that includes a publicly
+   available JS library and custom embedded script.
+2. When opening `report.html` in any view, it is served directly by S3 via a
```
Comment (akarve) [inline docs/Catalog/Preview.md:122]:
```suggestion
2. When opening `report.html` in any view it is served directly by S3 via a
```

File: docs/Catalog/Preview.md [hunk 5]
```diff
@@ -52,34 +52,83 @@ currently supported.
...
+1. `report.html` is a file in a bucket `example-bucket` that includes a publicly
+   available JS library and custom embedded script.
+2. When opening `report.html` in a bucket view, it is served directly by S3
+   via a signed HTTPS URL, e.g.
+   `https://example-bucket.s3.region.amazonaws.com/report.html?versionId=...&X-Amz-...`.
+3. All relative media and scripts are rendered in the same iframe relative-path
+   format:
+    * `./img.jpg` is resolved to `/img.jpg`
+    * `script.js` is resolved to `/script.js`
+4. `allow-same-origin` iframe sandbox token enabled,
```
Comment (akarve) [inline docs/Catalog/Preview.md:113]:
```suggestion
4. The `allow-same-origin` iframe sandbox token enabled,
```

File: docs/Catalog/Preview.md [hunk 6]
```diff
@@ -52,34 +52,83 @@ currently supported.
...
    format:
-    * `./img.jpg` is served as `/temporary-session-id/img.jpg`
-    * `script.js` is served as `/temporary-session-id/script.js`
-
-> **All files in the same package** are made temporarily publicly-available (for
-> lifetime of the session) under `/temporary-session-id`, even if not explicitly
-referenced in `report.html`.
+    * `./img.jpg` is resolved to `/temporary-session-id/img.jpg`
+    * `script.js` is resolved to `/temporary-session-id/script.js`
+5. `allow-same-origin` iframe sandbox token enabled,
```
Comment (akarve) [inline docs/Catalog/Preview.md:98]:
```suggestion
5. The `allow-same-origin` iframe sandbox token is enabled,
```

File: docs/Catalog/Preview.md [hunk 7]
```diff
@@ -52,34 +52,83 @@ currently supported.
...
+   the origin is the **Quilt Package File Server**,
+   LocalStorage API is available.
```
Comment (akarve) [inline docs/Catalog/Preview.md:100]:
```suggestion
   the LocalStorage API is available.
```

File: docs/Catalog/Preview.md [hunk 8]
```diff
@@ -52,34 +52,83 @@ currently supported.
...
+
+### Bucket view example with permissive rendering enabled
+
+1. `report.html` is a file in a bucket `example-bucket` that includes a publicly
+   available JS library and custom embedded script.
+2. When opening `report.html` in a bucket view, it is served directly by S3
```
Comment (akarve) [inline docs/Catalog/Preview.md:106]:
```suggestion
2. When opening `report.html` in bucket view, it is served directly by S3
```

File: docs/Catalog/Preview.md [hunk 9]
```diff
@@ -52,34 +52,83 @@ currently supported.
...
-2. Opening `report.html` generates  a new session `temporary-session-id`.
-3. The file is served from the iframe relative-path
-   `/temporary-session-id/report.html`.
+- Inside a package view with permissive rendering **enabled**:
+  the origin is the **Quilt Package File Server**.
+
+- Inside a bucket view with permissive rendering **enabled**:
+  the origin is the AWS S3 bucket endpoint.
+
+- With permissive rendering **disabled** (regardless of package vs bucket context):
```
Comment (akarve) [inline docs/Catalog/Preview.md:78]:
```suggestion
- With permissive rendering **disabled** (irrespective of package or bucket view):
```

**Review by akarve — APPROVED** [review]
A few suggestions in-line. One minor thing about the change set it looks to me like some of the GQL name spacing and method renaming could have been done in a separate PR.

**Comment by nl0** [comment]
> One minor thing about the change set it looks to me like some of the GQL name spacing and method renaming could have been done in a separate PR.

well, yes, i could've moved out the pre-factoring changes into a separate PR, but that would've slowed things down, and this PR turned out not that big anyways, even with the "unrelated" changes

File: catalog/app/components/Preview/loaders/Html/Html.tsx (hunk 1 above)
Comment (nl0) [reply catalog/app/components/Preview/loaders/Html/Html.tsx:217]: yep

File: catalog/app/components/Preview/renderers/IFrame.tsx (hunk 2 above)
Comment (nl0) [reply catalog/app/components/Preview/renderers/IFrame.tsx:15]: yep

**Review by fiskus — APPROVED** [review]

**Pull request merged**